File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    nums = {'one': '1', 'two': '2', 'three': '3', 'four': '4', 'five': '5', 'six': '6', 'seven': '7'
        , 'eight': '8', 'nine': '9', 'zero': '0'}

    lst = []
    res = []
    for i in range(len(s)):

        if s[i].isalpha():
            lst.append(s[i])
            tmp = ''.join(lst)

            if nums.get(tmp) :
                res.append(nums[tmp])
                lst = []
            else:
                continue

        else:

            res.append(str(s[i]))

    answer = int(''.join(res))

    return answer

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {
        String[] words = {"zero", "one", "two", "three", "four", 
                          "five", "six", "seven", "eight", "nine"};
        String answer = "";
        String tmp = "";
        
        for(char c:s.toCharArray()){
            if(Character.isDigit(c)){
                answer+=c;
            }else{
                tmp+=c;
                for(int i=0; i<words.length; i++){
                    if(tmp.equals(words[i])){
                        answer += i;
                        tmp="";
                    }
                }
            }  
        }
        
        return Integer.parseInt(answer);
    }
}